github/pkg/reconciler/source/controller.go

```diff
@@ -19,44 +19,53 @@ package source
 import (
 	"context"
 
-	"knative.dev/pkg/system"
-
-	"knative.dev/pkg/tracker"
+	"go.uber.org/zap"
 
 	"github.com/kelseyhightower/envconfig"
-
-	//k8s.io imports
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/tools/cache"
 
-	//Injection imports
-	sourcescheme "knative.dev/eventing-contrib/github/pkg/client/clientset/versioned/scheme"
-	githubinformer "knative.dev/eventing-contrib/github/pkg/client/injection/informers/sources/v1alpha1/githubsource"
-	ghreconciler "knative.dev/eventing-contrib/github/pkg/client/injection/reconciler/sources/v1alpha1/githubsource"
-	serviceclient "knative.dev/serving/pkg/client/injection/client"
-	kserviceinformer "knative.dev/serving/pkg/client/injection/informers/serving/v1/service"
-
-	//knative.dev/eventing imports
-	"knative.dev/eventing-contrib/github/pkg/apis/sources/v1alpha1"
-
-	//knative.dev/pkg imports
 	kubeclient "knative.dev/pkg/client/injection/kube/client"
 	"knative.dev/pkg/configmap"
 	"knative.dev/pkg/controller"
 	"knative.dev/pkg/logging"
 	"knative.dev/pkg/resolver"
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/tracker"
+
+	network "knative.dev/networking/pkg"
+
+	networkingclient "knative.dev/networking/pkg/client/injection/client"
+	networkinginformer "knative.dev/networking/pkg/client/injection/informers/networking/v1alpha1/ingress"
+	serviceclient "knative.dev/serving/pkg/client/injection/client"
+	kserviceinformer "knative.dev/serving/pkg/client/injection/informers/serving/v1/service"
+	"knative.dev/serving/pkg/reconciler/route/config"
+
+	"knative.dev/eventing-contrib/github/pkg/apis/sources/v1alpha1"
+	sourcescheme "knative.dev/eventing-contrib/github/pkg/client/clientset/versioned/scheme"
+	githubinformer "knative.dev/eventing-contrib/github/pkg/client/injection/informers/sources/v1alpha1/githubsource"
+	ghreconciler "knative.dev/eventing-contrib/github/pkg/client/injection/reconciler/sources/v1alpha1/githubsource"
 )
 
 // envConfig will be used to extract the required environment variables.
 // If this configuration cannot be extracted, then NewController will panic.
 type envConfig struct {
-	Image string `envconfig:"GH_RA_IMAGE"`
+	Image            string `envconfig:"GH_RA_IMAGE"`
+	EnableVanityURL  bool   `envconfig:"ENABLE_VANITY_URL"`
+	ServingNamespace string `envconfig:"SERVING_NAMESPACE"`
```

Comment:
This tells me we are doing something wrong.  The serving configmaps are not supported/public API, and by relying on them like this, you are inviting us to break you when we change it (either with realms or by switching to CRDs).

cc @tcnghia